pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -243,10 +244,16 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 				continue
 			}
 
-			routes, err := istio_route.BuildHTTPRoutesForVirtualService(node, push, virtualService, nameToServiceMap, port, map[string]bool{gatewayName: true})
-			if err != nil {
-				log.Debugf("%s omitting routes for virtual service %v/%v due to error: %v", node.ID, virtualService.Namespace, virtualService.Name, err)
-				continue
+			var routes []*route.Route
+			var exists bool
+			var err error
+			if routes, exists = gatewayRoutes[gatewayName]; !exists {
+				routes, err = istio_route.BuildHTTPRoutesForVirtualService(node, push, virtualService, nameToServiceMap, port, map[string]bool{gatewayName: true})
+				if err != nil {
+					log.Debugf("%s omitting routes for virtual service %v/%v due to error: %v", node.ID, virtualService.Namespace, virtualService.Name, err)
+					continue
+				}
+				gatewayRoutes[gatewayName] = routes
```

Comment:
Isn't the route releated to `virtualService`? Can not imagine how the ci passed

Reply:
thats a good point. Let me check

Reply:
Fixed. PTAL